test/Npgsql.Tests/SyncOrAsyncTestBase.cs

```diff
@@ -0,0 +1,24 @@
+using NUnit.Framework;
+
+namespace Npgsql.Tests
+{
+    [TestFixture(ExecutionMode.Synchronous)]
+    [TestFixture(ExecutionMode.Asynchronous)]
+    public abstract class SyncOrAsyncTestBase : TestBase
+    {
+        protected bool IsAsync => ExecutionMode == ExecutionMode.Asynchronous;
+
+        protected ExecutionMode ExecutionMode { get; }
+
+        protected SyncOrAsyncTestBase(ExecutionMode executionMode)
+        {
+            ExecutionMode = executionMode;
+        }
+    }
+
+    public enum ExecutionMode
```

Comment:
ExecutionMode is a very ambiguous name, why not SyncOrAsync? I'd also make the labels shorter (Sync and Async).